mne/io/curry/curry.py

```diff
@@ -0,0 +1,270 @@
+                 'TriggerOffsetUsec', 'DataFormat', 'SampleTimeUsec',
+                 'NUM_SAMPLES', 'NUM_CHANNELS', 'NUM_TRIALS', 'SAMPLE_FREQ_HZ',
+                 'TRIGGER_OFFSET_USEC', 'DATA_FORMAT', 'SAMPLE_TIME_USEC']
+
+
+    param_dict = dict()
+    unit_dict = dict()
+    with open(fname_base + INFO_FILE_EXTENSION[curry_vers]) as fid:
+        for line in iter(fid):
+            if any(var_name in line for var_name in var_names):
+                key, val = line.replace(" ", "").replace("\n", "").split("=")
+                param_dict[key.lower().replace("_", "")] = val
+            for type in CHANTYPES:
+                if "DEVICE_PARAMETERS" + CHANTYPES[type] + " START" in line:
+                    data_unit = next(fid)
+                    unit_dict[type] = data_unit.replace(" ", "").replace("\n", "").split("=")[-1]
+
+    for var in var_names[:7]:
+        if var.lower() not in param_dict:
+            raise KeyError("Variable %s cannot be found in the parameter file."
+                           % var)
+
+    n_samples = int(param_dict["numsamples"])
+    # n_ch = int(param_dict["numchannels"])
+    n_trials = int(param_dict["numtrials"])
+    sfreq = float(param_dict["samplefreqhz"])
+    offset = float(param_dict["triggeroffsetusec"]) * 1e-6
+    time_step = float(param_dict["sampletimeusec"]) * 1e-6
+    data_format = param_dict["dataformat"]
+
+    if (sfreq == 0) and (time_step != 0):
+        sfreq = 1. / time_step
+
+    # read labels from label files
+    labels = _read_curry_lines(fname_base + LABEL_FILE_EXTENSION[curry_vers],
+                               ["LABELS" + CHANTYPES[key] for key in CHANTYPES])
+
+    sensors = _read_curry_lines(fname_base + LABEL_FILE_EXTENSION[curry_vers],
+                                ["SENSORS" + CHANTYPES[key] for key in CHANTYPES])
+
+    all_chans = list()
+    for key in CHANTYPES:
+        for ind, chan in enumerate(labels["LABELS" + CHANTYPES[key]]):
+            ch = {"ch_name": chan,
+                  "unit": unit_dict[key],
+                  "kind": FIFFV_CHANTYPES[key]}
+            if key in ("meg", "eeg"):
+                loc = sensors["SENSORS" + CHANTYPES[key]][ind]
+                ch["loc"] = np.array(loc, dtype=float)
+
+            all_chans.append(ch)
+
+    ch_names = [chan["ch_name"] for chan in all_chans]
+    info = create_info(ch_names, sfreq)
+
+    # FIXME: Is it safer to use numpy.where(ch_dict["ch_name"] == all_chans["ch_name"]) or another method?
+    for ind, ch_dict in enumerate(info["chs"]):
+        ch_dict["unit"] = all_chans[ind]["unit"]
+        ch_dict["kind"] = all_chans[ind]["kind"]
+        if ch_dict["kind"] in (FIFF.FIFFV_MEG_CH,
+                               FIFF.FIFFV_EEG_CH):
+            ch_dict["loc"] = all_chans[ind]["loc"]
+
+        # TODO: This is extremely ugly. Is there a function to transform units automatically?
+        if ch_dict["unit"] == "uV":
+            ch_dict["cal"] = 1e-6
```

Comment:
can anybody help me out here? I need something that converts units like "uV" and "fT" automatically to the adequate ["cal"] value. Can this maybe be done using the FIFF unit types?

Reply:
Somewhere we might have a unit conversion thing, but I can't remember offhand. We use SI units so for now you can do:
```
units = dict(c=1e-2, m=1e-3, u=1e-6, μ=1e-6, n=1e-9, p=1e-12, f=1e-15)
ch_dict['cal'] = units[ch_dict['unit'][0]]
```

Basically for now follow what @agramfort usually advises

1. Make it work (use conditional branching or a dict, either is fine)
2. Make it pretty (see if we can use or refactor with existing code, if not then dict is probably the cleanest)
3. Make it fast (not too relevant here)

Reply:
`mne.defaults` has some useful stuff [here](https://github.com/mne-tools/mne-python/blob/master/mne/defaults.py#L14-L20) and [here](https://github.com/mne-tools/mne-python/blob/master/mne/defaults.py#L74) but I don't think it's actually set up to read a unit and spit out a scaling, so I guess @larsoner's suggestion wins.

Reply:
OK, thanks for that!